src/reanimated2/screenTransition/swipeSimulator.ts

```diff
@@ -0,0 +1,214 @@
+    return 1;
+  }
+  const elapsedTime = (_getAnimationTimestamp() - startingTimestamp) / 1000;
+  const currentPosition = velocity * elapsedTime;
+  const progress = currentPosition / distance;
+  return progress;
+}
+
+function easing(x: number): number {
+  'worklet';
+  return 1 - Math.pow(1 - x, 5);
+}
+
+function computeProgress(
+  screenTransitionConfig: ScreenTransitionConfig,
+  event: PanGestureHandlerEventPayload,
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  const screenWidth = screenTransitionConfig.screenDimensions.width;
+  const progressX = Math.abs(event.translationX / screenWidth);
+  const screenHeight = screenTransitionConfig.screenDimensions.height;
+  const progressY = Math.abs(event.translationY / screenHeight);
+  const progress = isTransitionCanceled
+    ? Math.max(progressX, progressY) / 2
+    : Math.max(progressX, progressY);
+  return progress;
+}
+
+function maybeScheduleNextFrame(
+  step: () => void,
+  isScreenReachDestination: boolean,
+  screenTransitionConfig: ScreenTransitionConfig,
+  event: PanGestureHandlerEventPayload,
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  if (!isScreenReachDestination) {
+    const stackTag = screenTransitionConfig.stackTag;
+    const progress = computeProgress(
+      screenTransitionConfig,
+      event,
+      isTransitionCanceled
+    );
+    global._manageScreenTransition(
+      ScreenTransitionCommand.Update,
+      stackTag,
+      progress
+    );
+    requestAnimationFrame(step);
+  } else {
+    if (screenTransitionConfig.onFinishAnimation) {
+      screenTransitionConfig.onFinishAnimation();
+    }
+  }
+}
+
+export function swipeSimulator(
+  event: PanGestureHandlerEventPayload,
+  screenTransitionConfig: ScreenTransitionConfig,
+  lockAxis?: 'x' | 'y' | undefined
+) {
+  'worklet';
+  const screenSize = screenTransitionConfig.screenDimensions;
+  const startingTimestamp = _getAnimationTimestamp();
+  const isTransitionCanceled = screenTransitionConfig.isTransitionCanceled;
+  const startingPosition = {
+    x: event.translationX,
+    y: event.translationY,
+  };
+  const direction = {
+    x: event.translationX > 0 ? 1 : -1,
+    y: event.translationY > 0 ? 1 : -1,
+  };
+  const finalPosition = isTransitionCanceled
+    ? { x: 0, y: 0 }
+    : { x: direction.x * screenSize.width, y: direction.y * screenSize.height };
+  const distance = {
+    x: Math.abs(finalPosition.x - startingPosition.x),
+    y: Math.abs(finalPosition.y - startingPosition.y),
+  };
+  const isScreenReachDestination = {
+    x: false,
+    y: false,
+  };
+  const BASE_VELOCITY = 300;
+  const velocity = { x: BASE_VELOCITY, y: BASE_VELOCITY };
+  if (lockAxis === 'x') {
+    velocity.y = 0;
+  } else if (lockAxis === 'y') {
+    velocity.x = 0;
+  } else {
+    if (Math.abs(startingPosition.x) > Math.abs(startingPosition.y)) {
+      velocity.x = BASE_VELOCITY;
+      velocity.y =
+        BASE_VELOCITY * Math.abs(startingPosition.y / startingPosition.x);
```

Comment:
Where is it all taken from?

Reply:
I have performed a simple calculation in order to construct a velocity vector that can move the screen directly to the correct corner. Without this calculation, the velocity vector would always be at a 45-degree angle.